### ISSUE TITLE
false positive `reportUnsafeMultipleInheritance` when `__init__`s call `super` safely

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#624]
# Well
```py
class A:
    def __init__(self, a: int, **kwargs: object):
        super().__init__(**kwargs)


class B:
    def __init__(self, b: int, **kwargs: object):
        super().__init__(**kwargs)

# OR
class A:
    def __init__(self, *, a: int, b: int):
        super().__init__(b=b)  # error: No parameter named "b"


class B:
    def __init__(self, *, b: int):
        super().__init__()

class Something(A, B):  # error: Multiple inheritance is not allowed because the following base classes contain `__init__` or `__new__` methods that may not get called: B
    ...
```

# So

This leads to the larger issue that there is no mechanism to safely forward args args around with super, leading to this constructor being unsafe:
```py
_ = Something(a=1, c=3)  # 'b' is missing, extra argument 'c'
```

i'm not sure how a solution would look, but maybe something along the lines of:

```py
from basedtyping import Super

class A:
    def __init__(self, a: int, *args: Super.args, **kwargs: Super.kwargs):
        super().__init__(*args, **kwargs)


class B:
    def __init__(self, b: int, *args: Super.args, **kwargs: Super.kwargs):
        super().__init__(*args, **kwargs)


class Something(A, B):
    ...

_ = Something(a=1, c=3)
#           ^?  (a: int, b: int) -> Something
```

i suppose this should also work with arbitrary methods

**Comment by DetachHead** [comment]
duplicate of #623?

**Comment by KotlinIsland** [comment]
> duplicate of #623?

no, this is a much more complicated scenario with no clear resolution imo

**Comment by DetachHead** [comment]
this just sounds like the exact reason i invented `reportUnsafeMultipleInheritance` in the first place. there's no way to statically check whether a class's subclass combines it with another class with an incompatible constructor so the best approach is to just ban it entirely.

i guess it might be possible to solve this fully if we add an "application" mode? (#539)
